### DISCUSSION TITLE
How can I make Renovate to PR only Node LTS versions in Dockerfile?

### DISCUSSION THREAD

**Discussion by tal-rofe** [renovatebot/renovate#14909] [Request Help] [answered]
I have the following Renovate configuration:
```js
{
	"extends": ["config:base", ":semanticCommitTypeAll(chore)", ":semanticCommitScopeDisabled"],
	"packageRules": [
		{
			"updateTypes": ["minor", "patch"],
			"automerge": true
		}
	],
	"timezone": "Asia/Jerusalem",
	"schedule": ["every weekend"]
}
```

I also have this line in my `Dockerfile`: `FROM node:16-alpine`

I get PR from Renovate to update it to `node:17-alpine`.
However, 17 is not LTS version, and 16 is the latest LTS.
I do want to accept major updates for Node as long as it is LTS one.

How can I do it?

**Comment by rarkins** [comment, marked as answer]
You'd need the regex manager to extract just the 16 part and then select "node" versioning

**Reply by rarkins** [reply]
If it were just 16 instead of 16-alpine then it would work out of the box

**Reply by tal-rofe** [reply]
@rarkins 

How about this one?
```
{
	"extends": ["config:base", ":semanticCommitTypeAll(chore)", ":semanticCommitScopeDisabled"],
	"enabledManagers": ["regex"],
	"regexMangers": [
		{
			"fileMatch": ["^Dockerfile$"],
			"matchStrings": ["FROM node:(?<currentValue>.*?)-alpine\\n"],
			"depNameTemplate": "node",
			"datasourceTemplate": "node"
		}
	],
	"packageRules": [
		{
			"updateTypes": ["minor", "patch"],
			"automerge": true
		}
	],
	"timezone": "Asia/Jerusalem",
	"schedule": ["every weekend"]
}
```

**Reply by rarkins** [reply]
You'll need a versioningTemplate=node in your regex manager too

**Reply by tal-rofe** [reply]
@rarkins 
This way?
```
{
	"$schema": "https://docs.renovatebot.com/renovate-schema.json",
	"extends": ["config:base", ":semanticCommitTypeAll(chore)", ":semanticCommitScopeDisabled"],
	"enabledManagers": ["regex"],
	"regexMangers": [
		{
			"fileMatch": ["^Dockerfile$"],
			"matchStrings": ["FROM node:(?<currentValue>.*?)-alpine\\n"],
			"depNameTemplate": "node",
			"datasourceTemplate": "node",
			"versioningTemplate": "node"
		}
	],
	"packageRules": [
		{
			"updateTypes": ["minor", "patch"],
			"automerge": true
		}
	],
	"timezone": "Asia/Jerusalem",
	"schedule": ["every weekend"],
	"stabilityDays": 7
}

```

**Reply by rarkins** [reply]
It should achieve the results you want, but you'll need to also disable the node dependency which the dockerfile manager finds

**Reply by tal-rofe** [reply]
@rarkins 
I am not sure whether the snippet above is correct.
I've set `"enabledManagers": ["regex"]` so according to the documentation it will disable **all other managers (including dockerfile**.

So why should I disable node dep in dockerfile manager?

**Reply by rarkins** [reply]
You're correct

**Reply by tal-rofe** [reply]
@rarkins 
So I do want dockerfile manager, how can I disable node dependency from this manager?

**Reply by rarkins** [reply]
Add a package rule with matchManagers=Dockerfile, matchPackageNames=node, and enable=false

**Reply by tal-rofe** [reply]
@rarkins 

Thanks, will check

**Reply by tal-rofe** [reply]
@rarkins 

Thank you, It solved it

**Reply by mschoettle** [reply]
Would you be able to share your final solution?

Also, just to confirm: Do I understand it right that it is only possible to configure it such that it does not update the major versions? So an update to the next LTS version would have to be done manually.

**Reply by tal-rofe** [reply]
@mschoettle 

Sure, here is my full configuration file:
```json
{
	"$schema": "https://docs.renovatebot.com/renovate-schema.json",
	"extends": ["config:base", ":semanticCommitTypeAll(chore)", ":semanticCommitScopeDisabled"],
	"regexManagers": [
		{
			"fileMatch": ["^Dockerfile$"],
			"matchStrings": ["FROM node:(?<currentValue>.*?)-alpine\\n"],
			"depNameTemplate": "node",
			"datasourceTemplate": "node",
			"versioningTemplate": "node"
		}
	],
	"packageRules": [
		{
			"updateTypes": ["minor", "patch"],
			"automerge": true
		},
		{
			"matchPackageNames": ["node"],
			"matchManagers": ["dockerfile"],
			"enabled": false
		}
	],
	"timezone": "Asia/Jerusalem",
	"schedule": ["every weekend"],
	"commitMessageAction": "🤖 Update"
}
```

Regarding your question- no. Renovate **WILL** update it automatically. Renovate handles correctly node LTS versions.
It just has issues with alpine ones. So to solve it, I "guided" Renovate to update the NODE ALPINE version as it would update it w/o ALPINE.

So clarify,, if I were to use only `FROM node:12` instead of `FROM node:12-alpine then Renovate will automatically update it to 16 (LTS). But using alpine makes issues. So I picked "manually" the version (12) itself and told Renovate to handle only this version.

**Reply by mschoettle** [reply]
Perfect, thanks a lot!
